src/libostree/ostree-fetcher-curl.c

```diff
@@ -335,7 +342,17 @@ check_multi_info (OstreeFetcher *fetcher)
       else
         {
           curl_easy_getinfo (easy, CURLINFO_RESPONSE_CODE, &response);
-          if (!is_file && !(response >= 200 && response < 300))
+
+          if (!is_file && response == 304 &&
+              (req->if_none_match != NULL || req->if_modified_since > 0))
+            {
+              /* Version on the server is unchanged from the version we have
+               * cached locally; report this as an out-argument, a zero-length
+               * response buffer, and no error. */
+              req->out_not_modified = TRUE;
+            }
+
+          if (!is_file && !(response >= 200 && response < 300) && response != 304)
```

Comment:
Minor: I think it'd be nice to match the libsoup version of this and use `else if` here. I like that it makes it really obvious that if the first branch is taken, it's not an error condition.

Reply:
I tried that and it breaks fetches if done in libcurl. Looking more closely, the structure of the code doesn’t match: in libsoup, the `if` block is entirely for error handling, but in libcurl, the first `if` block is for error handling and the subsequent `else if` and `else` blocks are all for success handling. We want to take one of those `else if`/`else` blocks even if `req->out_not_modified` has been set.

So I haven’t made this change.